ballerina/llm_client_azure_open_ai.bal

```diff
@@ -24,41 +25,104 @@ public type AzureOpenAIModelConfig record {|
+    "json_schema" 'type;
+    ResponseFormatJsonSchemaValue json_schema;
+|};
+
+type ResponseFormatJsonSchemaValue record {
+    string description?;
+    string name;
+    json schema?;
+    boolean? strict = true;
+};
+
+type ChatCompletionAzureResponse record {
+    AzureOpenAIResponseMessage[] choices?;
+};
+
+type AzureOpenAIResponseMessage record {
+    record {
+        string? content;
+    } message;
+};
+
 # Azure OpenAI model chat completion client.
 public isolated distinct client class AzureOpenAIModel {
     *Model;
 
-    private final chat:Client cl;
+    private final string serviceUrl;
     private final string deploymentId;
     private final string apiVersion;
+    private final string apiKey;
+    private final http:Client azureOpenAIClient;
 
-    public isolated function init(chat:Client|AzureOpenAIModelConfig azureOpenAI,
+    public isolated function init(AzureOpenAIModelConfig azureOpenAI,
             string deploymentId,
             string apiVersion) returns error? {
-        self.cl = azureOpenAI is chat:Client ?
-            azureOpenAI :
-            check new (azureOpenAI.connectionConfig, azureOpenAI.serviceUrl);
         self.deploymentId = deploymentId;
         self.apiVersion = apiVersion;
+        http:BearerTokenConfig|chat:ApiKeysConfig auth = azureOpenAI.connectionConfig.auth;
+
+        if auth is http:BearerTokenConfig {
+            self.apiKey = auth.token;
+        } else {
+            self.apiKey = auth.apiKey;
+        }
+
+        self.azureOpenAIClient = check new (
+            url = azureOpenAI.serviceUrl,
+            config = {
+                timeout: 60
+            }
+        );
     }
 
     isolated remote function call(string prompt, map<json> expectedResponseSchema) returns json|error {
-        chat:CreateChatCompletionRequest chatBody = {
-            messages: [{role: "user", "content": getPromptWithExpectedResponseSchema(prompt, expectedResponseSchema)}]
+        ChatCompletionAzureRequest chatBody = {
+            messages: [{role: "user", "content": getPromptWithExpectedResponseSchema(prompt, expectedResponseSchema)}],
+            response_format: check getJsonSchemaResponseTypeForAzureOpenAI(expectedResponseSchema)
         };
 
-        chat:Client cl = self.cl;
-        chat:CreateChatCompletionResponse chatResult =
-            check cl->/deployments/[self.deploymentId]/chat/completions.post(self.apiVersion, chatBody);
-        record {
-            chat:ChatCompletionResponseMessage message?;
-            chat:ContentFilterChoiceResults content_filter_results?;
-            int index?;
-            string finish_reason?;
-        }[]? choices = chatResult.choices;
+        string resourcePath = string `/deployments/${getEncodedUri(self.deploymentId)}/chat/completions?api-version=${self.apiVersion}`;
+        http:Request request = new;
+        json jsonBody = chatBody.toJson();
+        request.setPayload(jsonBody, "application/json");
```

Comment:
Did directly setting the payload not work?

Reply:
This is what `azure.chat` connector internally do.

Reply:
We don't have to do exactly that if there's a better way to do it.

Reply:
Still not resolved?